pulpcore/tests/functional/api/test_crd_artifacts.py

```
@@ -20,7 +20,12 @@ def pulpcore_random_file(tmp_path):
 
 def _do_upload_valid_attrs(pulpcore_bindings, file, data):
     """Upload a file with the given attributes."""
-    artifact = pulpcore_bindings.ArtifactsApi.create(str(file), **data)
+    # Create or get artifact if it already exists
+    try:
+        artifact = pulpcore_bindings.ArtifactsApi.create(str(file), **data)
+    except pulpcore_bindings.ApiException:
+        artifact = pulpcore_bindings.ArtifactsApi.list(sha256=data.get("sha256")).results[0]
```

Is this actually what we want?  If the goal is to test upload / artifact creation, you arguably *do not* want it to silently use already-created data, you want it to do the upload and fail if it cannot.  At least not unless the test itself wants to re-use data.

---

Would you prefer to delete artifacts/repositories if they already exist? Currently, if you run the tests multiple times on your local machine, these two of them fail. I would like to fix this behavior and I am okay with any solution that works.

Ideally, I would expect the tests to use some kind of fake or temporary storage so that repeated runs do not affect the results, but maybe there is a reason why the tests use the real storage.

---

Using random data is probably the best solution here.  It is admittedly much more difficult with other plugins.